src/models/AppState.js

```diff
@@ -260,7 +260,16 @@ function (Okta, Q, Factor, BrowserFeatures, Errors) {
         // user clicks 'deny' on his phone.
         deps: ['lastAuthResponse'],
         fn: function (res) {
-          return res.factorResult === 'REJECTED';
+          return res.factorResult === 'REJECTED' && !res.factorResultMessage;
+        }
+      },
+      'isMfaRejectedDueToOutdatedApp': {
```

Comment:
here's another idea: keep `isMfaRejectedByUser` (maybe change it to `isMfaRejected`) and add a new derived property to know if the app is outdated
```
isOktaVerifyAppOutdated: {
  deps: ['lastAuthResponse'],
  fn: function (res) {
    return res.factorResultMessage === 'OKTA_VERIFY_UPGRADE_REQUIRED';
  }
}
```

That way in the Form we listen to changes for `change:isMfaRejectedByUser` and we just handle the error message different, as opposed to adding a new listener. The logic for both listeners is basically the same, that is why I suggest this

With that said, what we have is good as well, it is just an idea

Reply:
I like this idea too but this introduces additional checks in the PushForm increasing the complexity a bit. Leaving it as it is.

Reply:
ok, I think it will probably make it simpler, but if we go with what we have today, the other change that is happening is that `isMfaRejectedByUser` is now checking for an empty `factorResultMessage`. Do we know if that is true for all rejected flows?
cause the new `isMfaRejectedDueToOutdatedApp` checks for an specific message, so if there is a flow that comes with a different messages, we won't be handling that anymore. Let's just double check that

Reply:
Good point Mau. I checked and isMfaRejectedByUser is true when factorResult is 'REJECTED' and factorResultMessage is empty. My check here works. 

But like you said I guess better to not bank on that I will make the check explicit.